src/encodings/decoding.rs

```diff
@@ -196,8 +237,17 @@ impl Decoder<BoolType> for PlainDecoder<BoolType> {
     assert!(self.bit_reader.is_some());
 
     let bit_reader = self.bit_reader.as_mut().unwrap();
-    let values_read = bit_reader.get_batch::<bool>(buffer, 1);
-    self.num_values -= values_read;
+    let mut values_read = bit_reader.get_batch::<bool>(buffer, 1);
+
+    // If the length of input buffer is larger than `num_values`, we may read trailing
```

Comment:
This happens when, say, the input `buffer` has length = 5, while `self.num_values` = 3. In this case, the `bit_reader.get_batch` will still try to fill the `buffer` with 5 elements. It may read the extra bits in the last trailing byte of its data, which is garbage. Therefore, we should avoid this.

I think a simpler way is to just pass `&mut buffer[..cmp::min(buffer.len(), self.num_values)]` instead of `buffer` to `get_batch`.